src/jobs/jobs.controller.ts

```diff
@@ -312,29 +372,75 @@ export class JobsController {
         });
       }),
     );
+    return jc.configVersion;
   }
 
 
   /**
    * Checking if user is allowed to create job according to auth field of job configuration
    */
-  // TBD
   async instanceAuthorization(
-    jobParams: Record<string, any> | undefined,
+    createJobDto: CreateJobDto,
     user: JWTUser,
+    functionType: string
   ): Promise<boolean> {
-    // Accepted options:
-    // #all, #datasetOwner, #datasetOwnerOrAccess, #authenticated,
+    const jc = await this.checkJobConfiguration(createJobDto.type);
 
-    if (jobParams === undefined) {
-      // TBD
-      return true;
-    } else if (jobParams.auth == JobsAuth.All) {
+    if (jc[functionType].auth == JobsAuth.All) {
+      // Public, no token required
       // nothing to do here
       return true;
-    } else if (jobParams.auth == JobsAuth.DatasetOwner) {
+    } else if (jc[functionType].auth == JobsAuth.Admin) {
+      // Verify that the user is an admin
+      if (user !== null && user.currentGroups.includes('admin')) {
+        throw new HttpException(
+          {
+            status: HttpStatus.UNAUTHORIZED,
+          },
+          HttpStatus.UNAUTHORIZED,
+        );
+      }
+      return true;
+    } else if (jc[functionType].auth == JobsAuth.Authenticated) {
+      // Verify that the user is authenticated
+      if (user === null) {
+        throw new HttpException(
+          {
+            status: HttpStatus.UNAUTHORIZED,
+          },
+          HttpStatus.UNAUTHORIZED,
+        );
+      }
+      return true;
+    } else if (jc[functionType].auth == JobsAuth.DatasetPublic) {
```

Comment:
It's not clear to me why `DatasetPublic` is needed. Isn't it covered by `DatasetAccess` already?

Reply:
I am not sure either. I covered all options provided in `jobs-auth.enum.ts`. We can discuss this again